app/jobs/milestone_event_job.rb

```diff
@@ -2,57 +2,81 @@ class MilestoneEventJob < ActiveJob::Base
 
   around_perform :guard_against_double_events
 
+  class_attribute :_action
+  class_attribute :_cache_key
+  class_attribute :_timestamp
+  class_attribute :_identifier
+
   def self.action(action)
-    @_action = action
+    self._action = action
   end
 
-  @_timestamp = Proc.new { Time.now.utc.iso8601 }
-  def self.timestamp(override=nil)
-    if override
-      @_timestamp = override
-    else
-      @_timestamp
-    end
+  self._timestamp = Proc.new { Time.now.utc.iso8601 }
+
+  def self.timestamp(override)
+    self._timestamp = override
   end
+
+  self._identifier = ->(p){ p['milestone']['number'] }
+
   def self.identifier(override=nil)
-    @_identifier = override
+    self._identifier = override
+  end
+
+  self._cache_key = ->(message) { "#{message[:meta][:action]}.#{message[:meta][:user]["login"]}.#{message[:meta][:identifier]}.#{message[:meta][:timestamp]}" }
+  
+  def self.cache_key(override)
+    self._cache_key = override
   end
 
   def self.build_meta(params)
-    milestone = params['milestone']
-    action = @_action.is_a?(String) ? @_action : @_action.call(params)
+    #TODO Fix this hack on remapping params to HWIA
+    params = HashWithIndifferentAccess.new(params)
+    milestone = HashWithIndifferentAccess.new(params['milestone'])
+    action = self._action.is_a?(String) ? self._action : self._action.call(params)
     HashWithIndifferentAccess.new(
       action: action,
-      timestamp: (@_timestamp || Proc.new{Time.now.utc.iso8601}).call(params),
+      timestamp: self._timestamp.call(params),
       correlationId: params['action_controller.params']['correlationId'],
       user: params['current_user'],
-      identifier: (@_identifier || ->(p){ p['milestone']['number']}).call(params),
+      identifier: self._identifier.call(params),
       repo_full_name: "#{milestone[:repo][:owner][:login]}/#{milestone[:repo][:name]}"
     )
   end
 
   def guard_against_double_events
-    payload = { meta:  self.class.build_meta(self.arguments.first) }
+    message = build_message(self.arguments.first)
     willPublish = true
+
     Rails.cache.with do |dalli|
-      key = "#{payload[:meta][:action]}.#{payload[:meta][:user]["login"]}.#{payload[:meta][:identifier]}.#{payload[:meta][:timestamp]}"
+      key = self._cache_key.call(message)
       willPublish = false if dalli.get(key)
-      dalli.set(key, payload.to_s)
+      Rails.logger.debug [self.class, 'ActiveJob:cache_key', key, 'willPublish', willPublish]
+      dalli.set(key, key)
```

Comment:
Should this be `dalli.set(key, message)`? Or does the set value not really matter since we're just checking if it's set?